cedar-policy/src/api.rs

```diff
@@ -1237,6 +1314,52 @@ pub enum SchemaError {
     /// Support for this escape form has been dropped.
     #[error("schema contained the non-supported `__expr` escape")]
     ExprEscapeUsed,
+    /// Errors parsing the natural schema syntax.
+    #[error("{0}")]
+    NaturalSyntaxError(#[from] NaturalSyntaxError),
+    /// IO errors when calling `from_reader`
+    #[error("I/O error while reading schmea: {0}")]
+    IOError(#[from] std::io::Error),
+    /// Error converting a into natural syntax
+    #[error("{0}")]
+    ToNaturalSyntax(#[from] ToNaturalSyntaxError),
+}
+
+/// Errors serializing Schemas to the natural syntax
+#[derive(Debug, Error, Diagnostic)]
+pub enum ToNaturalSyntaxError {
+    /// Duplicate names were found in the schema
+    #[error("There are type name collisions: [{}]", .0.iter().join(","))]
+    NameCollisions(NonEmpty<SmolStr>),
+}
+
+impl From<cedar_policy_validator::custom_schema::ToCustomSchemaStrError> for ToNaturalSyntaxError {
+    fn from(value: cedar_policy_validator::custom_schema::ToCustomSchemaStrError) -> Self {
+        match value {
+            cedar_policy_validator::custom_schema::ToCustomSchemaStrError::NameCollisions(
+                collisions,
+            ) => Self::NameCollisions(collisions),
+        }
+    }
+}
+
+/// Errors when parsing natural schema syntax
+#[derive(Debug, Diagnostic, Error)]
+pub enum NaturalSyntaxError {
+    /// Error Parsing a Schema in natural syntax
+    #[error("Error parsing schema: {0}")]
+    #[diagnostic(transparent)]
+    ParseError(cedar_policy_validator::custom_schema::parser::NaturalSyntaxParseErrors),
+}
+
+impl From<cedar_policy_validator::custom_schema::parser::NaturalSyntaxParseErrors>
+    for NaturalSyntaxError
+{
+    fn from(
+        value: cedar_policy_validator::custom_schema::parser::NaturalSyntaxParseErrors,
+    ) -> Self {
+        Self::ParseError(value)
+    }
 }
```

Comment:
instead of this I think we can just use `#[from]` in the declaration of `NaturalSyntaxError::ParseError`

Reply:
Fixed